extensions/data-plane/data-plane-http/src/main/java/org/eclipse/dataspaceconnector/dataplane/http/pipeline/HttpDataSource.java

```diff
@@ -0,0 +1,146 @@
+/*
+ *  Copyright (c) 2021 Microsoft Corporation
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Microsoft Corporation - initial API and implementation
+ *
+ */
+package org.eclipse.dataspaceconnector.dataplane.http.pipeline;
+
+
+import net.jodah.failsafe.RetryPolicy;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import org.eclipse.dataspaceconnector.dataplane.spi.pipeline.DataSource;
+import org.eclipse.dataspaceconnector.spi.EdcException;
+import org.eclipse.dataspaceconnector.spi.monitor.Monitor;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Objects;
+import java.util.stream.Stream;
+
+import static java.lang.String.format;
+import static net.jodah.failsafe.Failsafe.with;
+
+/**
+ * Pulls data from a source using an HTTP GET.
+ */
+public class HttpDataSource implements DataSource {
+    private String sourceEndpoint;
+    private String name;
+    private String requestId;
+    private RetryPolicy<Object> retryPolicy;
+    private OkHttpClient httpClient;
+    private Monitor monitor;
+
+    @Override
+    public Stream<Part> openStream() {
+        return Stream.of(getPart());
+    }
+
+    private HttpPart getPart() {
+        var request = new Request.Builder().url(sourceEndpoint + "/" + name).get().build();
```

Comment:
Hi @jimmarino 
For me this seems a bit oversimplified here:
 - a token can be requested to retrieve data from the API (e.g. Authorization header)
 - source API might also accept request parameters: `http://example.com?foo=bar`. You can find a possible approach to forward the parameters from the request here: https://github.com/microsoft/edc-showcase/blob/1847325e2d986bae6b900d00160575c91e1a97e2/extensions/transfer-http-proxy/src/main/java/org/eclipse/dataspaceconnector/transfer/httproxy/ForwardingController.java#L57

Reply:
OK, good point. Let me look into that.